src/windows/sid.rs

```diff
@@ -0,0 +1,153 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::{fmt::Display, str::FromStr};
+
+use winapi::{
+    shared::{
+        sddl::{ConvertSidToStringSidW, ConvertStringSidToSidW},
+        winerror::ERROR_INSUFFICIENT_BUFFER,
+    },
+    um::{
+        errhandlingapi::GetLastError,
+        securitybaseapi::{CopySid, EqualSid, GetLengthSid, IsValidSid},
+        winbase::{LocalFree, LookupAccountSidW},
+        winnt::{SidTypeUnknown, LPWSTR, PSID},
+    },
+};
+
+use crate::sys::utils::to_str;
+
+#[doc = include_str!("../../md_doc/sid.md")]
+#[derive(Clone, Debug)]
+pub struct Sid {
+    sid: Vec<u8>,
+}
+
+impl Sid {
+    /// Creates an `Sid` by making a copy of the given raw SID.
+    pub(crate) unsafe fn from_psid(psid: PSID) -> Option<Self> {
+        if psid.is_null() {
+            return None;
+        }
+
+        if IsValidSid(psid) == 0 {
+            return None;
+        }
+
+        let length = GetLengthSid(psid);
+
+        let mut sid = vec![0; length as usize];
+
+        if CopySid(length, sid.as_mut_ptr() as *mut _, psid) == 0 {
+            sysinfo_debug!("CopySid failed: {:?}", GetLastError());
+            return None;
+        }
+
+        Some(Self { sid })
+    }
+
+    /// Retrieves the account name of this SID.
+    pub fn account_name(&self) -> Option<String> {
+        unsafe {
+            let mut name_len = 0;
+            let mut domain_len = 0;
+            let mut name_use = SidTypeUnknown;
+
+            if LookupAccountSidW(
+                std::ptr::null_mut(),
+                self.sid.as_ptr() as *mut _,
+                std::ptr::null_mut(),
+                &mut name_len,
+                std::ptr::null_mut(),
+                &mut domain_len,
+                &mut name_use,
+            ) == 0
+            {
+                let error = GetLastError();
+                if error != ERROR_INSUFFICIENT_BUFFER {
+                    sysinfo_debug!("LookupAccountSidW failed: {:?}", error);
+                    return None;
+                }
+            }
+
+            let mut name = vec![0; name_len as usize];
+            let mut domain = vec![0; domain_len as usize];
+
+            if LookupAccountSidW(
+                std::ptr::null_mut(),
+                self.sid.as_ptr() as *mut _,
+                name.as_mut_ptr(),
+                &mut name_len,
+                domain.as_mut_ptr(),

```

Comment:
This argument is marked as optional so since it's not used after, I think you can pass `NULL` and not create a `Vec` for it.